infra/modules/providers/azure/service-principal/variables.tf

```diff
@@ -0,0 +1,26 @@
+variable "create_for_rbac" {
+  description = "Create a new Service Principle"
+  type        = "string"
+  default     = "true"
+}
+
+variable "service_principal_display_name" {
+  description = "Display name of the AD application"
+  type        = "string"
+}
+
+variable "service_principle_object_id" {
+  description = "Object Id of the service principle to assign a role"
+  type        = "string"
+  default     = ""
+}
+
+variable "role_name" {
+  description = "The name of the to assign a service principle too"
+  type        = "string"
+}
+
+variable "role_scope" {
```

Comment:
I may be incorrect on this but I believe the intended use for this module is to grant permissions to a variable number of scopes. This would more closely model the azure-cli as well, which supports multiple scopes. Given this, should we model this as a list?

Related question -- do we expect the role to change based on the scope? If so, then perhaps modeling this as a list is not the right approach. 

What are your thoughts?

Reply:
@TechnicallyWilliams and I talked about this also. The provider expects a single role/scope, so I went with that pattern. We could certainly broaden that to loop(=TF12) or simulate a loop (<TF11), but I'm happy with it this way at the moment.

Reply:
I believe that the looping pattern will be the same in both TF 0.11 and 0.12 as in both versions you'll need to use `count`. It will be something like this:
```hcl
resource "azurerm_role_assignment" "sp" {
  count                = "${length(var.role_scopes)}"
  role_definition_name = "${var.role_name}"
  principal_id         = "${var.create_for_rbac == "true" ? azuread_service_principal.sp.object_id : var.service_principle_object_id}"
  scope                = "${element(var.role_scopes, count.index)}"
}

...

variable "role_scope" {
  description = "..."
  type        = "list"
}
```

I think that extending beyond what the core providers expose as their APIs is a key benefit of using something like Cobalt. This would add some value IMO.

If this is a requirement that we'll need for the first milestone my suggestion is that we go ahead and add it now so that we can avoid refactoring it in the next set of PRs (i.e., once we add other modules where this will be used).

Reply:
Based on our convo, I'll let you decide the approach to take